stats/opentelemetry/client_metrics.go

```diff
@@ -123,8 +133,21 @@ func (csh *clientStatsHandler) HandleConn(context.Context, stats.ConnStats) {}
 
 // TagRPC implements per RPC attempt context management.
 func (csh *clientStatsHandler) TagRPC(ctx context.Context, info *stats.RPCTagInfo) context.Context {
+	// Numerous stats handlers can be used for the same channel. The cluster
+	// impl balancer which writes to this will only write once, thus have this
+	// stats handler's per attempt scoped context point to the same optional
+	// labels map if set.
+	var labels *istats.Labels
+	if labels = istats.GetLabels(ctx); labels == nil {
+		labels = &istats.Labels{
+			TelemetryLabels: make(map[string]string),
+		} // Create optional labels map only if first stats handler in possible chain for a channel.
+	}
+	ctx = istats.SetLabels(ctx, labels)
```

Comment:
Doesn't this need to be inside the `if`?

Do you have testing with multiple stats handlers to confirm they're working correctly?

Reply:
Moved. Will add a test in #7259.